android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -640,8 +640,8 @@ public void replaceAt(Object params)
 		}
 		if (children != null && children.size() > position) {
 			TiViewProxy childToRemove = children.get(position);
-			insertAt(params);
 			remove(childToRemove);
+			insertAt(params);
```

Comment:
Switched this around, as replacing child with existing child (changing position) would have remove the child completely.